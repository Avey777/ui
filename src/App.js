import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import SideNavContainer from 'src/side_nav';
import {
  publishNotification as publishNotificationAction,
  dismissNotification as dismissNotificationAction,
} from 'src/shared/actions/notifications';

const App = React.createClass({
  propTypes: {
    children: PropTypes.node.isRequired,
    location: PropTypes.shape({
      pathname: PropTypes.string,
    }),
    params: PropTypes.shape({
      sourceID: PropTypes.string.isRequired,
    }).isRequired,
    publishNotification: PropTypes.func.isRequired,
    dismissNotification: PropTypes.func.isRequired,
    notifications: PropTypes.shape({
      success: PropTypes.string,
      error: PropTypes.string,
      warning: PropTypes.string,
    }),
  },

  handleNotification({type, text}) {
    this.props.publishNotification(type, text);
  },

  handleDismissNotification(type) {
    this.props.dismissNotification(type);
  },

  render() {
    const {sourceID} = this.props.params;

    return (
      <div className="enterprise-wrapper--flex">
        <SideNavContainer sourceID={sourceID} addFlashMessage={this.handleNotification} currentLocation={this.props.location.pathname} />
        <div className="page-wrapper">
          {this.renderNotifications()}
          {this.props.children && React.cloneElement(this.props.children, {
            addFlashMessage: this.handleNotification,
          })}
        </div>
      </div>
    );
  },

  renderNotifications() {
    const {success, error} = this.props.notifications;
    if (!success && !error) {
      return null;
    }
    if (success) {
      return (
        <div className="flash-messages">
          <div className="alert alert-success" role="alert">
            {success}{this.renderDismiss()}
          </div>
        </div>
      );
    }
    if (error) {
      return (
        <div className="flash-messages">
          <div className="alert alert-danger" role="alert">
            {error}{this.renderDismiss()}
          </div>
        </div>
      );
    }
  },

  renderDismiss() {
    return (
      <button className="close" data-dismiss="alert" aria-label="Close" onClick={this.handleDismissNotification}>
        <span className="icon remove"></span>
      </button>
    );
  },
});

function mapStateToProps(state) {
  return {
    notifications: state.notifications,
  };
}

export default connect(mapStateToProps, {
  publishNotification: publishNotificationAction,
  dismissNotification: dismissNotificationAction,
})(App);
