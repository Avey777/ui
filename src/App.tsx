import React, {SFC, ReactChildren} from 'react'

import RightClickLayer from 'src/clockface/components/right_click_menu/RightClickLayer'
import Nav from 'src/pageLayout'
import Notifications from 'src/shared/components/notifications/Notifications'

interface Props {
  children: ReactChildren
}

const App: SFC<Props> = ({children}) => (
  <div className="chronograf-root">
    <Notifications />
    <RightClickLayer />
    <Nav />
    {children}
  </div>
)

export default App
