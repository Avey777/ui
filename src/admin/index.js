import UsersPage from './containers/UsersPage';
export {UsersPage};
