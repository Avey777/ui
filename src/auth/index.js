import Login from './Login'
import Purgatory from './Purgatory'

import {
  UserIsAuthenticated,
  Authenticated,
  UserIsNotAuthenticated,
} from './Authenticated'
export {
  Login,
  Purgatory,
  UserIsAuthenticated,
  Authenticated,
  UserIsNotAuthenticated,
}
