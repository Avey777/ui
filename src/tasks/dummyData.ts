import {Task, TaskStatus} from 'src/types/v2/tasks'

export const dummyTasks: Task[] = [
  {
    id: '02ef9deff2141000',
    organizationId: '02ee9e2a29d73000',
    name: 'pasdlak',
    status: TaskStatus.Active,
    owner: {id: '02ee9e2a19d73000', name: ''},
    flux:
      'option task = {\n  name: "pasdlak",\n  cron: "2 0 * * *"\n}\nfrom(bucket: "inbucket") \n|> range(start: -1h)',
    cron: '2 0 * * *',
    organization: {
      links: {
        buckets: '/api/v2/buckets?org=RadicalOrganization',
        dashboards: '/api/v2/dashboards?org=RadicalOrganization',
        log: '/api/v2/orgs/02ee9e2a29d73000/log',
        members: '/api/v2/orgs/02ee9e2a29d73000/members',
        self: '/api/v2/orgs/02ee9e2a29d73000',
        tasks: '/api/v2/tasks?org=RadicalOrganization',
      },
      id: '02ee9e2a29d73000',
      name: 'RadicalOrganization',
    },
  },
  {
    id: '02f12c50dba72000',
    organizationId: '02ee9e2a29d73000',
    name: 'somename',
    status: TaskStatus.Active,
    owner: {id: '02ee9e2a19d73000', name: ''},
    flux:
      'option task = {\n  name: "somename",\n  every: 1m,\n}\nfrom(bucket: "inbucket") \n|> range(start: -task.every)',
    every: '1m0s',
    organization: {
      links: {
        buckets: '/api/v2/buckets?org=RadicalOrganization',
        dashboards: '/api/v2/dashboards?org=RadicalOrganization',
        log: '/api/v2/orgs/02ee9e2a29d73000/log',
        members: '/api/v2/orgs/02ee9e2a29d73000/members',
        self: '/api/v2/orgs/02ee9e2a29d73000',
        tasks: '/api/v2/tasks?org=RadicalOrganization',
      },
      id: '02ee9e2a29d73000',
      name: 'RadicalOrganization',
    },
  },
]
