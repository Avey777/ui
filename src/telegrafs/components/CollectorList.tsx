// Libraries
import React, {PureComponent} from 'react'
import {connect} from 'react-redux'
import _ from 'lodash'

// Components
import {IndexList} from 'src/clockface'
import CollectorRow from 'src/telegrafs/components/CollectorRow'

// Types
import {ITelegraf as Telegraf} from '@influxdata/influx'
import {Sort} from '@influxdata/clockface'
import {SortTypes} from 'src/shared/selectors/sort'
import {AppState} from 'src/types'

//Utils
import {getDeep} from 'src/utils/wrappers'

// Selectors
import {getSortedResource} from 'src/shared/selectors/sort'

type SortKey = keyof Telegraf

interface OwnProps {
  collectors: Telegraf[]
  emptyState: JSX.Element
  onDelete: (telegraf: Telegraf) => void
  onUpdate: (telegraf: Telegraf) => void
  onOpenInstructions: (telegrafID: string) => void
  onFilterChange: (searchTerm: string) => void
  sortKey: string
  sortDirection: Sort
  sortType: SortTypes
  onClickColumn: (nextSort: Sort, sortKey: SortKey) => void
}

interface StateProps {
  sortedCollectors: Telegraf[]
}

type Props = OwnProps & StateProps

class CollectorList extends PureComponent<Props> {
  public state = {
    sortedCollectors: this.props.sortedCollectors,
  }

  componentDidUpdate(prevProps) {
    const {collectors, sortedCollectors, sortKey, sortDirection} = this.props

    if (
      prevProps.sortDirection !== sortDirection ||
      prevProps.sortKey !== sortKey ||
      prevProps.collectors.length !== collectors.length
    ) {
      this.setState({sortedCollectors})
    }
  }

  public render() {
    const {emptyState, sortKey, sortDirection, onClickColumn} = this.props

    return (
      <>
        <IndexList>
          <IndexList.Header>
            <IndexList.HeaderCell
              sortKey={this.headerKeys[0]}
              sort={sortKey === this.headerKeys[0] ? sortDirection : Sort.None}
              columnName="Name"
              width="50%"
              onClick={onClickColumn}
            />
            <IndexList.HeaderCell columnName="Bucket" width="25%" />
            <IndexList.HeaderCell columnName="" width="25%" />
          </IndexList.Header>
          <IndexList.Body columnCount={3} emptyState={emptyState}>
            {this.collectorsList}
          </IndexList.Body>
        </IndexList>
      </>
    )
  }

  private get headerKeys(): SortKey[] {
    return ['name']
  }

  public get collectorsList(): JSX.Element[] {
    const {
      collectors,
      onDelete,
      onUpdate,
      onOpenInstructions,
      onFilterChange,
    } = this.props
    const {sortedCollectors} = this.state

    if (collectors !== undefined) {
      return sortedCollectors.map(collector => (
        <CollectorRow
          key={collector.id}
          collector={collector}
          bucket={getDeep<string>(collector, 'plugins.0.config.bucket', '')}
          onDelete={onDelete}
          onUpdate={onUpdate}
          onOpenInstructions={onOpenInstructions}
          onFilterChange={onFilterChange}
        />
      ))
    }
  }
}

const mstp = (state: AppState, props: OwnProps): StateProps => {
  return {
    sortedCollectors: getSortedResource(state.telegrafs.list, props),
  }
}

export default connect<StateProps, {}, OwnProps>(mstp)(CollectorList)
