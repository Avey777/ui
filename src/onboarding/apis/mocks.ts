import {telegrafConfigsResponse, authResponse} from 'mocks/dummyData'

const telegrafsGet = jest.fn(() => Promise.resolve(telegrafConfigsResponse))
const authorizationsGet = jest.fn(() => Promise.resolve(authResponse))

export const telegrafsAPI = {
  telegrafsGet,
}

export const authorizationsAPI = {
  authorizationsGet,
}
