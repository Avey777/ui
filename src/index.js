import React from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {Router, Route, useRouterHistory} from 'react-router'
import {createHistory} from 'history'
import {syncHistoryWithStore} from 'react-router-redux'

import App from 'src/App'
import AlertsApp from 'src/alerts'
import CheckSources from 'src/CheckSources'
import {HostsPage, HostPage} from 'src/hosts'
import {KubernetesPage} from 'src/kubernetes'
import {Login, UserIsAuthenticated, Authenticated, UserIsNotAuthenticated} from 'src/auth'
import {KapacitorPage, KapacitorRulePage, KapacitorRulesPage, KapacitorTasksPage} from 'src/kapacitor'
import DataExplorer from 'src/data_explorer'
import {DashboardsPage, DashboardPage} from 'src/dashboards'
import {CreateSource, SourcePage, ManageSources} from 'src/sources'
import {AdminPage} from 'src/admin'
import NotFound from 'src/shared/components/NotFound'
import configureStore from 'src/store/configureStore'
import {getMe, getSources} from 'shared/apis'
import {authRequested, authReceived, meRequested, meReceived} from 'shared/actions/auth'
import {disablePresentationMode} from 'shared/actions/app'
import {publishNotification} from 'shared/actions/notifications'
import {loadLocalStorage} from './localStorage'

import 'src/style/chronograf.scss'

import {HTTP_FORBIDDEN, HEARTBEAT_INTERVAL} from 'shared/constants'

const rootNode = document.getElementById('react-root')

let browserHistory
const basepath = rootNode.dataset.basepath
window.basepath = basepath
if (basepath) {
  browserHistory = useRouterHistory(createHistory)({
    basename: basepath, // this is written in when available by the URL prefixer middleware
  })
} else {
  browserHistory = useRouterHistory(createHistory)({
    basename: "",
  })
}

const store = configureStore(loadLocalStorage(), browserHistory)
const {dispatch} = store

browserHistory.listen(() => {
  dispatch(disablePresentationMode())
})

window.addEventListener('keyup', (event) => {
  if (event.key === 'Escape') {
    dispatch(disablePresentationMode())
  }
})

const history = syncHistoryWithStore(browserHistory, store)

const Root = React.createClass({
  getInitialState() {
    return {
      hasDeterminedAuth: false,
    }
  },

  componentWillMount() {
    this.checkAuth()
  },

  activeSource(sources) {
    const defaultSource = sources.find((s) => s.default)
    if (defaultSource && defaultSource.id) {
      return defaultSource
    }
    return sources[0]
  },

  async checkAuth() {
    dispatch(authRequested())
    dispatch(meRequested())
    try {
      await this.startHeartbeat({shouldDispatchResponse: true})
    } catch (error) {
      console.error(error)
    }
  },

  // TODO: use once auth router is working
  redirectFromRoot(_, replace, callback) {
    getSources().then(({data: {sources}}) => {
      if (sources && sources.length) {
        const path = `/sources/${this.activeSource(sources).id}/hosts`
        replace(path)
      }
      callback()
    })
  },

  async startHeartbeat({shouldDispatchResponse}) {
    try {
      const {data: me, auth} = await getMe()
      if (shouldDispatchResponse) {
        dispatch(authReceived(auth))
        dispatch(meReceived(me))
      }

      setTimeout(this.startHeartbeat.bind(null, {shouldDispatchResponse: false}), HEARTBEAT_INTERVAL)
    } catch (error) {
      if (error.auth) {
        dispatch(authReceived(error.auth))
        dispatch(meReceived(null))
      }
      if (error.status === HTTP_FORBIDDEN) {
        dispatch(publishNotification('error', 'Session timed out. Please login again.'))
      } else {
        dispatch(publishNotification('error', 'Cannot communicate with server.'))
      }
    }
  },

  render() {
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route path="/" component={UserIsAuthenticated(CreateSource)/* add back onEnter={this.redirectFromRoot} */} />
          <Route path="login" component={UserIsNotAuthenticated(Login)} />
          <Route path="sources/new" component={UserIsAuthenticated(CreateSource)} />
          <Route path="sources/:sourceID" component={UserIsAuthenticated(App)}>
            <Route component={UserIsAuthenticated(CheckSources)}>
              <Route component={Authenticated}>
                <Route path="manage-sources" component={ManageSources} />
                <Route path="manage-sources/new" component={SourcePage} />
                <Route path="manage-sources/:id/edit" component={SourcePage} />
                <Route path="chronograf/data-explorer" component={DataExplorer} />
                <Route path="hosts" component={HostsPage} />
                <Route path="hosts/:hostID" component={HostPage} />
                <Route path="kubernetes" component={KubernetesPage} />
                <Route path="kapacitor-config" component={KapacitorPage} />
                <Route path="kapacitor-tasks" component={KapacitorTasksPage} />
                <Route path="alerts" component={AlertsApp} />
                <Route path="dashboards" component={DashboardsPage} />
                <Route path="dashboards/:dashboardID" component={DashboardPage} />
                <Route path="alert-rules" component={KapacitorRulesPage} />
                <Route path="alert-rules/:ruleID" component={KapacitorRulePage} />
                <Route path="alert-rules/new" component={KapacitorRulePage} />
                <Route path="admin" component={AdminPage} />
              </Route>
            </Route>
          </Route>
          <Route path="*" component={NotFound} />
        </Router>
      </Provider>
    )
  },
})

if (rootNode) {
  render(<Root />, rootNode)
}
