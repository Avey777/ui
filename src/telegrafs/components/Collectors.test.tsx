// Libraries
import React from 'react'
import {renderWithRedux} from 'src/mockState'

// Components
import CollectorList from 'src/telegrafs/components/CollectorList'

// Constants
import {telegraf} from 'mocks/dummyData'

const setup = (override?) => {
  const props = {
    collectors: telegraf,
    emptyState: <></>,
    onDelete: jest.fn(),
    onUpdate: jest.fn(),
    onOpenInstructions: jest.fn(),
    onOpenTelegrafConfig: jest.fn(),
    ...override,
  }

  const wrapper = renderWithRedux(<CollectorList {...props} />)

  return {wrapper}
}

describe('CollectorList', () => {
  describe('rendering', () => {
    it('renders', () => {
      const {wrapper} = setup()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
