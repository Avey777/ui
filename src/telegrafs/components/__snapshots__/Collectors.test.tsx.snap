// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectorList rendering renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="index-list"
      >
        <thead
          class="index-list--header"
        >
          <tr>
            <th
              class="index-list--header-cell index-list--align-left index-list--sortable"
              style="width: 50%;"
            >
              Name
              <span
                class="index-list--sort-arrow"
              >
                <span
                  class="icon caret-down"
                />
              </span>
            </th>
            <th
              class="index-list--header-cell index-list--align-left"
              style="width: 25%;"
            >
              Bucket
            </th>
            <th
              class="index-list--header-cell index-list--align-left"
              style="width: 25%;"
            >
              
            </th>
          </tr>
        </thead>
        <tbody
          class="index-list--empty"
        >
          <tr
            class="index-list--empty-row"
          >
            <td
              colspan="3"
            >
              <div
                class="index-list--empty-cell"
                data-testid="empty-state"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="index-list"
    >
      <thead
        class="index-list--header"
      >
        <tr>
          <th
            class="index-list--header-cell index-list--align-left index-list--sortable"
            style="width: 50%;"
          >
            Name
            <span
              class="index-list--sort-arrow"
            >
              <span
                class="icon caret-down"
              />
            </span>
          </th>
          <th
            class="index-list--header-cell index-list--align-left"
            style="width: 25%;"
          >
            Bucket
          </th>
          <th
            class="index-list--header-cell index-list--align-left"
            style="width: 25%;"
          >
            
          </th>
        </tr>
      </thead>
      <tbody
        class="index-list--empty"
      >
        <tr
          class="index-list--empty-row"
        >
          <td
            colspan="3"
          >
            <div
              class="index-list--empty-cell"
              data-testid="empty-state"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
